pkgs/applications/version-management/git-and-tools/git-my/default.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "git-my";
+  version = "1.1.2";
+
+  src = fetchFromGitHub {
+    owner = "davidosomething";
+    repo = "git-my";
+    rev = version;
+    sha256 = "0jji5zw25jygj7g4f6f3k0p0s9g37r8iad8pa0s67cxbq2v4sc0v";
+  };
+
+  dontBuild = true;
+
+  installPhase = ''
+    install -Dm755 -t "$out"/bin ./git-my
+  '';
+
+  meta = let inherit (stdenv) lib; in {
```

Comment:
```suggestion
  meta = with stdenv.lib; {
```

Reply:
For these comments, I have the impression that Nix was moving away from `with` syntax from various conversations here? [The papercut thread discusses it masking the source of variables without evaulation](https://discourse.nixos.org/t/the-papercut-thread-post-your-small-annoyances-confusions-here/3100/6?u=bb010g), NixOS/nix#3042 dislikes it from a gradual typing perspective, and NixOS/nix#1361's discussion includes multiple people in favor of deprivation. I've been avoiding it in my new expressions since learning about it.

Are there places where it's more or less expected in Nixpkgs? (E.g. in a derivation's `meta` attribute vs. a new chunk of significant abstraction.)

Reply:
@bb010g It's likely because probably 90% of the expressions for packages in nixpkgs look like this.
For more complicated code you should wary of https://discourse.nixos.org/t/the-papercut-thread-post-your-small-annoyances-confusions-here/3100/6?u=bb010g. This is isolated to `meta` so it's fine.